tiledb/appl/blob_array/test/unit_blob_array_main.cc

```diff
@@ -0,0 +1,607 @@
+    REQUIRE(wb_ba.is_open() == true);
+  };
+
+  tiledb::sm::Datatype value_type;
+  uint32_t value_size;
+  const void* ptr_value;
+  CHECK(wb_ba.get_file_ext(&value_type, &value_size, &ptr_value).ok() == false);
+  CHECK(
+      wb_ba.get_mime_type(&value_type, &value_size, &ptr_value).ok() == false);
+  CHECK(
+      wb_ba.get_mime_encoding(&value_type, &value_size, &ptr_value).ok() ==
+      false);
+  open_wbba_for_write();
+  CHECK(wb_ba.get_file_ext(&value_type, &value_size, &ptr_value).ok() == false);
+  CHECK(
+      wb_ba.get_mime_type(&value_type, &value_size, &ptr_value).ok() == false);
+  CHECK(
+      wb_ba.get_mime_encoding(&value_type, &value_size, &ptr_value).ok() ==
+      false);
+  open_wbba_for_read();
+  CHECK(wb_ba.get_file_ext(&value_type, &value_size, &ptr_value).ok() == true);
+  CHECK(wb_ba.get_mime_type(&value_type, &value_size, &ptr_value).ok() == true);
+  CHECK(
+      wb_ba.get_mime_encoding(&value_type, &value_size, &ptr_value).ok() ==
+      true);
+
+  REQUIRE(blob_array->close().ok() == true);
+
+  auto show_dir = [&](std::string path) {
+    std::vector<tiledb::sm::URI> filelist;
+    tiledb::sm::URI uri_path(path);
+    vfs_->vfs_->ls(uri_path, &filelist);
+    std::cout << "path " << path << ", nitems " << filelist.size() << std::endl;
+    for (auto afile : filelist) {
+      uint64_t fsize = 0;
+      vfs_->vfs_->file_size(afile, &fsize);
+      std::cout << afile.to_path() << " " << fsize << std::endl;
+    }
+  };
+  auto show_dirs = [&]() -> void {
+    show_dir(localfs_temp_dir_);
+    show_dir(test_array_name);
+#if _WIN32
+    auto separator = "\\";
+#else
+    auto separator = "/";
+#endif
+    std::cout << "...__fragments..." << std::endl;
+    show_dir(test_array_name + separator + "__fragments");
+    std::cout << "...__meta..." << std::endl;
+    show_dir(test_array_name + separator + "__meta");
+  };
+  // compare actual contents for equality
+  auto cmp_files_check = [&](const std::string& file1,
+                             const std::string& file2) -> bool {
+    std::stringstream cmp_files_cmd;
+#ifdef _WIN32
+    // 'FC' does not like forward slashes
+    cmp_files_cmd << "FC "
+                  << tiledb::sm::path_win::slashes_to_backslashes(
+                         // tiledb::sm::path_win::path_from_uri(csv_path))
+                         tiledb::sm::path_win::path_from_uri(file1))
+                  << " "
+                  << tiledb::sm::path_win::slashes_to_backslashes(
+                         tiledb::sm::path_win::path_from_uri(file2))
+                  << " > nul";
+#else
+    // tiledb::sm::VFS::abs_path(output_path) << " > nul";
+    cmp_files_cmd << "diff " << file1 << " " << file2 << " > nul";
+#endif
+    // auto result = !system(cmpfilescmd.str().c_str());
+    auto result = system(cmp_files_cmd.str().c_str()) == 0;
+    CHECK(result);
+
+    if (!result) {
+      std::cout << "cmp " << file1 << "," << file2 << "different." << std::endl;
```

Comment:
remove

Reply:
removed (but was useful for quicker diagnosis when error encountered of items involved or in event error may not immediately reproduce)